spec/grape/validations/validators/default_spec.rb

```diff
@@ -99,6 +114,28 @@ def app
     expect(last_response.body).to eq({ root: params }.to_json)
   end
 
+  it 'does not allows faulty optional arrays' do
+    params = { some_things:
+                 [
+                   { foo: 'one', options: [{ name: 'wat', value: 'nope' }] },
+                   { foo: 'two', options: [{ name: 'wat' }] },
+                   { foo: 'three' }
+                 ] }
+    get '/nested_optional_array', root: params
+    expect(last_response.status).to eq(400)
```

Comment:
Test that the actual response contains the correct value being wrong. Maybe break this one up more, too.

Reply:
I don't understand

Reply:
In this case the status can be 400 for all kinds of reasons. Check `last_response.body` that it says something about the parameter being incorrect.

Reply:
got it 👍